### PR TITLE
Re-apply "Add assert that the root widget has been attached"

### PR DESCRIPTION
**Pull request by tvolkert**
This re-applies #32437 with a fix to the broken device lab test.

### CODE REVIEW

**Comment by goderbauer** [comment]
Looks like cirrus is rather unhappy here?

File: packages/flutter_driver/lib/src/common/message.dart [hunk 1]
```diff
@@ -32,6 +32,21 @@ abstract class Command {
...
+  /// This defaults to true to force the application under test to call [runApp]
+  /// before attempting to remotely drive the application. Subclasses may
+  /// override this to return false if they allow invocation before the
+  /// application has started.
+  ///
+  /// See also:
+  ///
+  ///  * [WidgetsBinding.isRootWidgetAttached], which indicates whether the
+  ///    widget tree has been initialized.
+  @protected
```
Comment (goderbauer) [inline packages/flutter_driver/lib/src/common/message.dart]: This probably shouldn't be protected if you want to use it in extension.dart?

Comment (tvolkert) [reply packages/flutter_driver/lib/src/common/message.dart]: Oops - done

**Review by goderbauer — APPROVED** [review]
LGTM

**Pull request merged**